packages/keycloak-user-management/src/ducks/userGroups.ts

```diff
@@ -0,0 +1,227 @@
+
+/**
+ * Gets name from KeycloakUsersFilters
+ *
+ * @param _ - the redux store
+ * @param props - the user groups filters object
+ */
+export const getGroupName = (_: Partial<Store>, props: Filters) => props.name;
+
+/**
+ * Gets searchText from filter props
+ *
+ * @param _ - the redux store
+ * @param props - the user groups filters object
+ */
+export const getSearchText = (_: Partial<Store>, props: Filters) => props.searchText;
+
+/** returns all user groups in the store as values whose keys are their respective ids
+ *
+ * @param {any} state - the redux store
+ * @returns {object} - users object as values, respective ids as keys
+ */
+export function getKeycloakUserGroupsById(
+  state: Partial<Store>
+): { [key: string]: KeycloakUserGroup } {
+  return (state as Dictionary)[reducerName].userGroupsById;
+}
+
+/** gets keycloak users as an array of user objects
+ *
+ * @param {any} state - the redux store
+ * @returns {Array} - an array of users objs
+ */
+export function getKeycloakUserGroupsArray(state: Partial<Store>): KeycloakUserGroup[] {
+  return values(getKeycloakUserGroupsById(state));
+}
+
+export const getUserGroupByName = () =>
+  /**
+   * Gets all groups whose name includes phrase given in name filter prop
+   *
+   * @returns {Array} - groups whose name includes phrase given in name filter prop
+   */
+  createSelector(getKeycloakUserGroupsArray, getGroupName, (userGroupsArray, name) =>
+    name
+      ? userGroupsArray.filter((userGroup: KeycloakUserGroup) =>
+          userGroup.name.toLowerCase().includes(name.toLowerCase())
+        )
+      : userGroupsArray
+  );
+
+export const getKeycloakUserGroupsByIds = () =>
+  /**
+   * Gets all user groups whose identifiers appear in ids filter prop value
+   *
+   * @returns {Array} - users whose identifiers appear in ids filter prop value
+   */
+  createSelector(
+    getKeycloakUserGroupsById,
+    getUserGroupsIds,
+    getKeycloakUserGroupsArray,
+    (userGroupsById, ids, userGroupsArray) => {
+      if (ids === undefined) {
+        return userGroupsArray;
+      }
+      if (ids.length > 0) {
+        return ids.map((id: string) => userGroupsById[id]);
+      }
+      return [];
+    }
+  );
+
+const userGroupsByIdsSelector = getKeycloakUserGroupsByIds();
+const userGroupsByNameSelector = getUserGroupByName();
+
+export const makeKeycloakUserGroupsSelector = () =>
+  /** practitioner array selector factory
+   * aggregates response from all applied filters and returns results
+   *
+   * @returns {Array} - aggregates response from all applied filters and returns results
+   */
+  createSelector(
+    userGroupsByIdsSelector,
+    userGroupsByNameSelector,
+    getSearchText,
+    (arr1, arr2, searchText) => {
+      const unfilteredResults = intersect([arr1, arr2], JSON.stringify);
+      const matchesUserName = (userGroup: KeycloakUserGroup) => {
+        if (!searchText) {
+          return true;
+        }
```

Comment:
how would this hold up if searchText was an empty string, I think its best to be explicti in this check i.e `searchText === undefined`